src/repr/src/scalar.rs

```diff
@@ -3083,8 +3157,12 @@ impl<'a> ScalarType {
             },
             ScalarType::Date,
             ScalarType::Time,
-            ScalarType::Timestamp,
-            ScalarType::TimestampTz,
+            ScalarType::Timestamp {
```

Comment:
This and the tz variant should also get a `MIN_PRECISION` and `None` variants.

Reply:
How are these values used? Is this for testing as well?

Reply:
Added!